test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,978 @@
+          } catch (error) {
+            expect(error).to.be.an.instanceOf(Error);
+            if (error instanceof ValidationError) {
+              expect(error.get('id')[0].message).to.equal('ID must be an integer!');
+            }
+          }
+        });
+      });
+    });
+
+    describe('pass all paths when validating', async () => {
+      const vars = beforeEach2(async () => {
+        class Project extends Model<InferAttributes<Project>, InferCreationAttributes<Project>> {
+          static associations: {
+            users: HasMany<Project, Task>,
+          };
+
+          declare name: CreationOptional<string>;
+        }
+
+        Project.init({
+          name: {
+            type: DataTypes.STRING,
+            allowNull: false,
+            defaultValue: 'unknown',
+            validate: {
+              isIn: [['unknown', 'hello', 'test']],
+            },
+          },
+        }, { sequelize });
+
+        class Task extends Model<InferAttributes<Task>, InferCreationAttributes<Task>> {
+          static associations: {
+            users: BelongsTo<Task, Project>,
+          };
+
+          declare something: number;
+        }
+
+        Task.init({
+          something: DataTypes.INTEGER,
+        }, { sequelize });
+
+        Project.hasOne(Task);
+        Task.belongsTo(Project);
+
+        await Project.sync({ force: true });
+        await Task.sync({ force: true });
+
+        return { Project, Task };
+      });
+
+      it('produce 3 errors', async () => {
+        try {
+          await vars.Project.create({});
+        } catch (error) {
+          expect(error).to.be.an.instanceOf(Error);
+          if (error instanceof ValidationError) {
+            delete error.stack; // longStackTraces
+            expect(error.errors).to.have.length(3);
+          }
+        }
+      });
+    });
+
+    describe('not null schema validation', async () => {
+      const vars = beforeEach2(async () => {
+        class Project extends Model<InferAttributes<Project>, InferCreationAttributes<Project>> {
+          declare name: CreationOptional<string>;
+        }
+
+        Project.init({
+          name: {
+            type: DataTypes.STRING,
+            allowNull: false,
+            defaultValue: 'unknown',
+            validate: {
+              isIn: [['unknown', 'hello', 'test']],
+            },
+          },
+        }, { sequelize });
+
+        return { Project };
+      });
+
+      it('correctly throws an error using create method ', async () => {
+        await vars.Project.create({})
+          .then(() => {
+            throw new Error('Validation must be failed');
+          }, async () => {
+            // fail is ok
+          });
+      });
+
+      it('correctly throws an error using create method with default generated messages', async () => {
+        try {
+          await vars.Project.create({});
+        } catch (error) {
+          if (error instanceof ValidationError) {
```

Comment:
this must be asserted, otherwise test will pass if a random error was thrown when it _must_ be a ValidationError